tasks/update-cnb-dependency/main.go

```diff
@@ -58,7 +59,12 @@ func updateCNBDependencies() error {
 		return errors.Wrap(err, "failed to construct list of dependencies to add")
 	}
 
-	originalDeps := config.BuildpackTOML.Metadata.Dependencies.ExpandByStack()
+	var deps Dependencies
+	err = mapstructure.WeakDecode(config.BuildpackTOML.Metadata[DependenciesKey], &deps)
```

Comment:
Why are we only using WeakDecode here? Is it to convert the version into a string?